packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -318,13 +338,29 @@ export class TrimergeClient<
     this.clientList = Array.from(this.clientMap.values());
     this.clientListSubs.emitChange(event);
   }
-  private sync(): void {
+  private async sync() {
     const commits = this.unsyncedCommits;
     if (commits.length > 0 || this.newPresence !== undefined) {
       this.unsyncedCommits = [];
       this.updateSyncState({ localSave: 'saving' });
-      this.store.update(commits, this.newPresence);
+      try {
+        await this.store.update(
+          commits.map((c) => c.commit),
+          this.newPresence,
+        );
+      } catch (e) {
+        for (const commit of commits) {
+          if (commit.reject) {
+            commit.reject(e);
+          }
+        }
+      }
       this.newPresence = undefined;
+      for (const commit of commits) {
+        if (commit.resolve) {
+          commit.resolve();
```

Comment:
While it's technically ok (i.e., a no-op) to resolve a promise immediately after rejecting it (on line 354), I think it makes the code a little harder to reason about.

Reply:
This shouldn't be necessary anymore based on your point above about awaiting sync.